dev/tools/test-migration.py

```
@@ -564,7 +630,10 @@ def validate_migration(active_pod_info):
         
         # Verify the storedVersions has been successfully pruned to just v1beta1
         res = run_cmd(["kubectl", "get", "crd", crd, "-o", "jsonpath={.status.storedVersions}"], capture_output=True)
-        stored_versions = json.loads(res.stdout)
+        try:
+            stored_versions = json.loads(res.stdout)
+        except json.JSONDecodeError as e:
+            raise AssertionError(f"Failed to parse storedVersions for CRD {crd}. Output: {res.stdout}. Error: {e}")
```

Ruff (0.15.18)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)